esp/platform/espcontext.cpp

```diff
@@ -547,11 +574,26 @@ class CEspContext : public CInterface, implements IEspContext
         m_txSummary.clear();
     }
 
+    virtual void setTxSummaryProfile(ITxSummaryProfile* txProfile)
+    {
+        m_txSummaryProfile.set(txProfile);
```

Comment:
Why is the summary profile attached to the context and not the summary itself? Any entity can use the summary interface directly, and some can only use the summary interface. Would it introduce problems if the summary held the profile reference and applied the profile when needed? This could allow the summary to be more data complete prior to the flushing the context. I'm thinking about the possibility of including the "current" summary content in an ESP response (which is a feature that has been requested) without assuming which summary interface is being used.

Reply:
Ok good idea. I'll switch it.